services/facility_queue/facility_queue.py

```diff
@@ -56,63 +55,74 @@ async def load_from_db(self):
-                        commit_photometry(
-                            json_response,
-                            altdata,
-                            followup_request.id,
-                            instrument.id,
-                            followup_request.requester.id,
-                        )
-                        req.status = 'complete'
-                        session.add(req)
-
-                    elif json_response['starttimestamp']:
-                        followup_request.status = f"Task is running (started at {json_response['starttimestamp']})"
-                        await self.put(req)
-                    else:
-                        followup_request.status = f"Waiting for job to start (queued at {json_response['timestamp']})"
-                        await self.put(req)
-                else:
-                    followup_request.status = f'error: {response.content}'
+            dt = datetime.utcnow() - req.last_query
+            if dt < WAIT_TIME_BETWEEN_QUERIES:
+                print(
+                    f"Holding {req.id} for {(WAIT_TIME_BETWEEN_QUERIES-dt).seconds} seconds"
+                )
+                await self.put(req)
+            else:
+                session = Session()
+                followup_request = session.query(FollowupRequest).get(
+                    req.followup_request_id
+                )
+                instrument = followup_request.allocation.instrument
+                altdata = followup_request.allocation.altdata
+
+                response = request_session.request(
+                    req.method,
+                    req.endpoint,
+                    json=req.data,
+                    params=req.params,
+                    headers=req.headers,
+                )
 
-                session.add(followup_request)
-                session.commit()
+                if instrument.name == "ATLAS":
+                    from skyportal.facility_apis.atlas import commit_photometry
+
+                    if response.status_code == 200:
+                        try:
+                            json_response = response.json()
+                        except Exception:
+                            raise ('No JSON data returned in request')
+
+                        if json_response['finishtimestamp']:
+                            req.status = "Committing photometry to database"
+                            commit_photometry(
+                                json_response,
+                                altdata,
+                                followup_request.id,
+                                instrument.id,
+                                followup_request.requester.id,
+                            )
+                            req.status = 'complete'
+                            session.add(req)
+                            print(f"Request {req.id} completed")
+
+                        elif json_response['starttimestamp']:
```

Comment:
With all these clauses, just think about that one item in the queue scenario, so that we don't churn up the DB or the output terminal.

Reply:
yeah I made fewer of them.